datafusion/expr/src/expr.rs

```diff
@@ -1869,10 +1869,14 @@ mod test {
         let exp2 = col("a") + lit(2);
         let exp3 = !(col("a") + lit(2));
 
-        assert!(exp1 < exp2);
-        assert!(exp2 > exp1);
-        assert!(exp2 > exp3);
-        assert!(exp3 < exp2);
+        // Since comparisons are done using hash value of the expression
```

Comment:
hm, but does that actually mean the comparison itself works not correctly sometimes?

Reply:
I think the calculation is consistent (will always be *either* < or >)

However, I wonder if the hash values are always consistent (like perhaps do they vary on x86_64 and M1 platforms?) Does that matter 🤔

Reply:
Ahash is not a stable hash algorithm and may not only change between platforms, but also minor releases. I lack context into why we are ordering based on hashes, but my initial response is this is probably incorrect, especially if it is inconsistent with the equality relation.

Reply:
> but my initial response is this is probably incorrect, especially if it is inconsistent with the equality relation.

Why is it "inconsistent with equality" ? If two exprs are equal, they would have the same hash, so the order between hash values would be consistent (on a certain platform and release)

That is not to say we shouldn't change how ordering is done, but I just don't understand this comment

Reply:
I haven't checked this, but if the PartialEq implementation isn't also using this hash approach, the two would be inconsistent